lib/IonRequest.ts

```diff
@@ -79,6 +75,169 @@ export default class IonRequest {
+    const delta = {
+      patches,
+      updateCommitment: nextUpdateCommitmentHash
+    };
+
+    const deltaHash = Multihash.canonicalizeThenHashThenEncode(delta, hashAlgorithmInMultihashCode);
+    const nextRecoveryCommitmentHash = Multihash.canonicalizeThenDoubleHashThenEncode(input.nextRecoveryPublicKey, hashAlgorithmInMultihashCode);
+
+    const signedDataPayloadObject = {
+      recoveryCommitment: nextRecoveryCommitmentHash,
+      recoveryKey: recoveryPublicKey,
+      deltaHash: deltaHash
+    };
+
+    const compactJws = await secp256k1.ES256K.sign(
+      signedDataPayloadObject,
+      input.recoveryPrivateKey,
+      { alg: 'ES256K' }
+    );
+
+    return {
+      type: OperationType.Recover,
+      didSuffix: input.didSuffix,
+      revealValue: revealValue,
+      delta: delta,
+      signedData: compactJws
+    };
+  }
+
+  public static async createUpdateRequest (input: {
+    didSuffix: string;
+    updatePrivateKey: JwkEs256k;
+    nextUpdatePublicKey: JwkEs256k;
+    servicesToAdd?: IonServiceModel[];
+    idsOfServicesToRemove?: string[];
+    publicKeysToAdd?: IonPublicKeyModel[];
+    idsOfPublicKeysToRemove?: string[];
+  }): Promise<IonUpdateRequestModel> {
+    const patches = [];
+    // Create patches for add services
+    const servicesToAdd = input.servicesToAdd;
+    if (servicesToAdd !== undefined && servicesToAdd.length > 0) {
+      const patch = {
+        action: PatchAction.AddServices,
+        services: servicesToAdd
+      };
+
+      patches.push(patch);
+    }
+
+    // Create patches for remove services
+    const idsOfServicesToRemove = input.idsOfServicesToRemove;
+    if (idsOfServicesToRemove !== undefined && idsOfServicesToRemove.length > 0) {
+      const patch = {
+        action: PatchAction.RemoveServices,
+        ids: idsOfServicesToRemove
+      };
+
+      patches.push(patch);
+    }
+
+    // Create patches for adding public keys
+    const publicKeysToAdd = input.publicKeysToAdd;
+    if (publicKeysToAdd !== undefined && publicKeysToAdd.length > 0) {
+      const patch = {
+        action: PatchAction.AddPublicKeys,
+        publicKeys: publicKeysToAdd
+      };
+
+      patches.push(patch);
+    }
+
+    // Create patch for removing public keys
+    const idsOfPublicKeysToRemove = input.idsOfPublicKeysToRemove;
+    if (idsOfPublicKeysToRemove !== undefined && idsOfPublicKeysToRemove.length > 0) {
+      const patch = {
+        action: PatchAction.RemovePublicKeys,
+        ids: idsOfPublicKeysToRemove
+      };
+
+      patches.push(patch);
+    }
+
+    const updatePublicKey = this.getPublicKeyFromPrivateKey(input.updatePrivateKey);
+    const hashAlgorithmInMultihashCode = IonSdkConfig.hashAlgorithmInMultihashCode;
+    const revealValue = Multihash.canonicalizeThenHashThenEncode(updatePublicKey, hashAlgorithmInMultihashCode);
+
+    const nextUpdateCommitmentHash = Multihash.canonicalizeThenDoubleHashThenEncode(input.nextUpdatePublicKey, hashAlgorithmInMultihashCode);
```

Comment:
Sanity: need to call `IonRequest.validateEs256kOperationPublicKey(...)` on `nextUpdatePublicKey`?